doc/array_buffer.md

```diff
@@ -1,5 +1,127 @@
-# Array buffer
+# ArrayBuffer
 
-You are reading a draft of the next documentation and it's in continuos update so
-if you don't find what you need please refer to:
-[C++ wrapper classes for the ABI-stable C APIs for Node.js](https://nodejs.github.io/node-addon-api/)
+The `ArrayBuffer` class corresponds to the JavaScript `ArrayBuffer` class.
+
+## Methods
+
+### New
+
+Allocates a new `ArrayBuffer` object with a given length.
+
+```cpp
+static ArrayBuffer New(napi_env env, size_t byteLength);
+```
+
+- `[in] env`: The environment in which to create the ArrayBuffer object.
+- `[in] byteLength`: The length to be allocated, in bytes.
+
+Returns a new `ArrayBuffer` object.
+
+### New
+
+Wraps the provided external data into a new `ArrayBuffer` object.
+
+The `ArrayBuffer` object does not assume ownership for the data and expects it
+to be valid for the lifetime of the object. Since the `ArrayBuffer` is subject
+to garbage collection this overload is only suitable for data which is static
+and never needs to be freed.
```

Comment:
I like your description here.  

I know we discussed doing a free, but thinking about it I don't like us trying to free an object that we don't know was allocated with malloc or new.

What would be good to figure out though is how to avoid conversion errors from NaN which result in leaks.  I'd be tempted to remove or deprecate this one and force callers to indicate they don't need finalization by passing a null to one of the other methods.  Since the wrapper is not part of the abi contract I think we could delete the method and provide instructions on what to use instead.

Reply:
Maybe just deprecate as opposed to remove with the doc saying "don't use"

Reply:
My only concern about "doc deprecated" is that it requires people to see the doc when converting their code.